hw/ip/prim/README.md

```diff
@@ -4,96 +4,156 @@
 
-To allow hardware blocks to be generic across implementations of *technology-dependent primitives*, a feature of FuseSoC called *[virtual cores][]* is used.
-Any implementation of a virtual core must provide the same functionality and interface.
-The module name must be the same, and the parameter and port lists must include all the required members of the interface.
-For an analogy, virtual cores can be thought of as similar to virtual methods in C++ or SystemVerilog.
-A hardware block can depend on a virtual core, which can then be substituted for a *concrete* implementation by the user.
+[Fusesoc]: https://github.com/olofk/fusesoc
+[Virtual Cores]: https://fusesoc.readthedocs.io/en/stable/user/build_system/virtual_cores.html
 
-To find the interface of a given virtual primitive look at its generic implementation in `hw/prim_generic/`.
-Taking `hw/ip/prim_generic/prim_generic_flop_2sync.core` as an example, you'll see the following at the head of the file.
+All technology-dependent primitives in OpenTitan use Virtual Cores to allow them to be substitued at build-time.
+For this reason, they are also referred to as ***Virtual Primitives***.
+
+FuseSoC Virtual Cores work by adding additional metadata to a Core, marking it as an implementation of another 'virtual' Core.
+The Virtual Core is a unique *[VLNV][]*, and does not exist as a named `.core` file in the tree.
+Taking `hw/ip/prim_generic/prim_generic_flop_2sync.core` as an example, you will see the following at the head of the file:
+
+[VLNV]: https://fusesoc.readthedocs.io/en/stable/user/build_system/core_files.html#the-core-name-version-and-description
 
 ```yaml
+CAPI=2:
 name: "lowrisc:prim_generic:flop_2sync"
 description: "Generic implementation of a flop-based synchronizer"
 virtual:
   - lowrisc:prim:flop_2sync
 ```
+Descriptively, we might say that the virtual core VLNV is `lowrisc:prim:flop_2sync`, and one possible implementation of this virtual core is `lowrisc:prim_generic:flop_2sync`.
+As a virtual primitive, the interface (module name and ports) for `flop_2sync` is common among all implementations, and hence all instantiations.
+
+By depending only on the virtual VLNV of a virtual primative, cores become generic over that primitive.
+Any core that implements this virtual core may be chosen at build time to provide the implementation for all instantiations.
+During build time, all virtual cores must be resolved to a concrete implementation.
+If a virtual core cannot be resolved because no implementations are found or specified at build-time, a "conflicting-requirements" error will be generated by the FuseSoC solver.
+The following section further describes the resolution process.
+
+It is still possible and valid for a Core to depend only on a specific implementation of a technology-dependent primitive, however the implementation cannot be substituted at build time in this case, and it functions as a normal FuseSoC Core.
```

Comment:
Is it worth adding a note here about what happens if I have two different implementations of a virtual core that both get requested directly?